File: final project angular (1)/src/app/model/review.ts
import { Reviewreplay } from './reviewreplay';

export class Review {
    public  User_Id?:string;
      public  Vendor_Id?:string;
      public  Comment?:string;
      public  catagory_id?:Number;
      public  username?:string;
      public  vendorname?:string;
      public  catname?:string;
      public  PostDate?:Date;
      public  ID?:Number;
      public liked?:boolean;
      public  replies:Reviewreplay[];
      public user:string  
}

File: final project core/final project core/Final project/Models/our tables/Review.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Final_project.Models.OurIdentity
{
    public class Review
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }
        [Required(ErrorMessage ="Please Enter The Date")]
        //  [DisplayFormat(DataFormatString = "{0:dd.MM.yyyy}")]
        [DataType(DataType.Date)]
        public DateTime PostDate { get; set; }
        [MaxLength(250)][MinLength(20)]
        [Required(ErrorMessage ="Comment Is Required")]
        public string Comment { get; set; }
        [ForeignKey("ApplicationUser")]
        public string User_Id { get; set; }
     
        [ForeignKey("Vendor")]
        public string Vendor_Id { get; set; }
        [ForeignKey("catagory")]
        public int catagory_id { get; set; }

        public bool liked { get; set; }
        public catagory catagory { get; set; }
        public ApplicationUser ApplicationUser { set; get; }
        public ApplicationUser Vendor { set; get; }
        

    }
}
